### ISSUE TITLE
Workload Install Type Can be Incorrect with Manifests Authored for Younger SDKs

### ISSUE THREAD

**Issue by nagilson** [dotnet/sdk#39275]
### Describe the bug
Running `dotnet workload --info` can sometimes report the incorrect install type. For example, it may say that MSI based workloads are `FileBased` even though the SDK is running under MSI mode and those workloads are installed via MSIs.

![image](https://github.com/dotnet/sdk/assets/23152278/58496490-4122-4d80-9561-175327d45d3d)

### To Reproduce

You need to have an SDK with a version that owns the manifests but is different from the Manifest Version. 

What happens is that the SDK looks for an MSI sentinel file under `metadata` in `program files` using the Manifest Version, such as 8.0.100. But if the SDK version is different, such as 8.0.200, the MSI sentinel file will be in a folder called `8.0.200`, not a folder with the manifest version `8.0.100`.

The result of the lookup will be whatever 8.0.100 is, whether or not the SDK that owns the manifests (8.0.200) is file based, or MSI based. In this example, because the 8.0.100 folder in metadata does not exist, it will report the MSI workloads installed by the MSI SDK (8.0.200) to be file based, which is incorrect. However, if 8.0.200 is file based but uses 8.0.100 version manifests, and there is an SDK that's 8.0.100 and MSI based on the system, I'd imagine this bug would work in the opposite direction too.

# Proposal

Perhaps we should have an msi file for each workload, or update this logic some other way.

**Comment by dsplaisted** [comment]
We are using the workload manifest feature band where we should be using the feature band of the current SDK.  Here is the line where this is happening:

https://github.com/dotnet/sdk/blob/main/src/Cli/dotnet/commands/dotnet-workload/WorkloadCommandParser.cs#L91

We should change that to this:

```c#
reporter.WriteLine($"       {WorkloadInstallType.GetWorkloadInstallType(new SdkFeatureBand(Utils.Product.Version), dotnetPath).ToString(),align}"
```

**Comment by vdanche** [comment]
@Forgind This issue will not have a fix in net8.0, right? still repro on 8.0.402 sdk
![image](https://github.com/user-attachments/assets/c4d14837-4ca5-4bc2-9655-6373ed67d707)

FYI @marcpopMSFT

**Comment by Forgind** [comment]
The PR to fix this was merged into main last week, not release/8.0.4xx, so it won't be fixed until 9.0

**Comment by vdanche** [comment]
@Forgind checked with 9.0.100-rc.2.24453.20 exe install, 
The install type for maui is displayed correctly, but install type for aspire is dislayed "FileBased", **is it expected?**
![image](https://github.com/user-attachments/assets/29639ae7-d7c4-4210-960b-5dcfffbecc1b)

**Comment by dsplaisted** [comment]
It looks like this was merged to main after we had branched for 10.0.  I've opened a PR to backport it to 9.0: #43231

**Comment by vdanche** [comment]
Checked with  9.0.100-rc.2.24468.2 SDK exe install, install type for aspire is dislayed as "Msi" now
![image](https://github.com/user-attachments/assets/1cdbed01-cab3-4807-b99b-6d6ef1f2c1b2)
